internal/tiltfile/files.go

```diff
@@ -80,6 +80,19 @@ func (s *tiltfileState) execLocalCmd(t *starlark.Thread, cmd model.Cmd, logOutpu
 		c.Stderr = io.MultiWriter(stderr, logOutput)
 	}
 
+	// if Tilt was invoked with `tilt up --port=XXXXX`, local() calls to use the Tilt API will fail due to trying to
+	// connect to the default port, so explicitly populate the TILT_PORT environment variable if it isn't already
+	hasTiltPort := false
+	for _, e := range c.Env {
+		if strings.HasPrefix(e, "TILT_PORT") {
```

Comment:
should this be HasPrefix(e, "TILT_PORT=") ? note the '='

Reply:
Good call - tweaked that when making it more generic to handle `TILT_HOST` so it'll need to be `${key}=` in f3d9caa